src/transformers/models/instructblipvideo/modular_instructblipvideo.py

```diff
@@ -103,6 +103,10 @@ class InstructBlipVideoConfig(PretrainedConfig):
     ```"""
 
     model_type = "instructblipvideo"
+    sub_configs = ["text_config", "qformer_config", "vision_config"]
+    text_config_class = "AutoConfig"
+    qformer_config_class = "InstructBlipVideoQFormerConfig"
+    vision_config_class = "InstructBlipVideoVisionConfig"
```

Comment:
Hi! Super nice work :fire: I was wondering about this, for composite modular-based _new_ models, could we add something in the converter to enforce this structure?

Reply:
Sounds good having it automated, I'll try to enable it through this PR if I can find a way

Reply:
maybe with AST? we have typed args for the configs that inform of which config goes where, too

Reply:
yes, seems to be doable if you enforce config docstring to contain the sub-config class in type-hinting. Maybe we'll add it as the next PR so we can take time for making modular better. Also I've another PR on improving modular in general, might as well go there as a new feature

Reply:
cc @Cyrilvallez since you are working closely with modular. Do you think this is smth we can add in your PR or maybe subsequent PRs?

Reply:
Hey, sorry for the late reply! This is definitely do-able - it would require parsing the `__init__` to check subconfig-specific assignments and add them as class attributes in the body.
However, it breaks a bit the "no magic" policy of the modular, as it means we automatically add stuff based on simple naming conventions.